@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -526,18 +646,23 @@ export class VisibleTileSet {
     }
 
     /**
+     * Computes the list of tiles that are to be rendered on screen.
+     *
      * Search cache to replace visible but yet empty tiles with already loaded siblings in nearby
      * zoom levels.
      *
      * Useful, when zooming in/out and when "newly elected" tiles are not yet loaded. Prevents
      * flickering by rendering already loaded tiles from upper/higher zoom levels.
      */
-    private fillMissingTilesFromCache() {
+    private computeTilesToRender() {
```

Comment:
I actually found the old name more descriptive, no?

Reply:
The old name is misleading IMO and caused some confusion until I had read the code, because this function computes the list of tiles to be rendered, as opposed to just filling missing tiles (which it also does yes).
Maybe one idea is to split this into two functions, one called computeTilesToRender, and another fillMissingTilesFromCache which takes the output of the first function and fills in the gaps, that could make the code a bit easier to read.

Reply:
Still open